TiKV assertion panic related to CDC

## Bug Report

<!-- Thanks for your bug report! Don't worry if you can't fill out all the sections. -->

### What version of TiKV are you using?

v8.5.1

### What operating system and CPU are you using?
<!-- If you're using Linux, you can run `cat /proc/cpuinfo` -->

### Steps to reproduce
<!-- If possible, provide a recipe for reproducing the error. A complete runnable program is good. -->

### What did you expect?

### What did happened?


```json
{"level":"INFO","caller":"raft.rs:2684","message":"switched to configuration","time":"2025/05/29 02:44:22.563 +00:00","raft_id":12700241,"region_id":12700240,"thread_id":156,"config":"Configuration { voters: Configuration { incoming: Configuration { voters: {24087147, 12700242, 12700241} }, outgoing: Configuration { voters: {} } }, learners: {}, learners_next: {}, auto_leave: false }"}
{"level":"FATAL","caller":"lib.rs:479","message":"assertion `left == right` failed\n  left: TimeStamp(458355287523328275)\n right: TimeStamp(458355287523328316)","time":"2025/05/29 02:44:22.737 +00:00","thread_id":47,"backtrace":"   0: tikv_util::set_panic_hook::{{closure}}\n             at /workspace/source/tikv/components/tikv_util/src/lib.rs:478:18\n   1: <alloc::boxed::Box<F,A> as core::ops::function::Fn<Args>>::call\n             at /root/.rustup/toolchains/nightly-2023-12-28-aarch64-unknown-linux-gnu/lib/rustlib/src/rust/library/alloc/src/boxed.rs:2029:9\n      std::panicking::rust_panic_with_hook\n             at /root/.rustup/toolchains/nightly-2023-12-28-aarch64-unknown-linux-gnu/lib/rustlib/src/rust/library/std/src/panicking.rs:783:13\n   2: std::panicking::begin_panic_handler::{{closure}}\n             at /root/.rustup/toolchains/nightly-2023-12-28-aarch64-unknown-linux-gnu/lib/rustlib/src/rust/library/std/src/panicking.rs:657:13\n   3: std::sys_common::backtrace::__rust_end_short_backtrace\n             at /root/.rustup/toolchains/nightly-2023-12-28-aarch64-unknown-linux-gnu/lib/rustlib/src/rust/library/std/src/sys_common/backtrace.rs:171:18\n   4: rust_begin_unwind\n             at /root/.rustup/toolchains/nightly-2023-12-28-aarch64-unknown-linux-gnu/lib/rustlib/src/rust/library/std/src/panicking.rs:645:5\n   5: core::panicking::panic_fmt\n             at /root/.rustup/toolchains/nightly-2023-12-28-aarch64-unknown-linux-gnu/lib/rustlib/src/rust/library/core/src/panicking.rs:72:14\n   6: core::panicking::assert_failed_inner\n   7: core::panicking::assert_failed\n             at /root/.rustup/toolchains/nightly-2023-12-28-aarch64-unknown-linux-gnu/lib/rustlib/src/rust/library/core/src/panicking.rs:297:5\n   8: cdc::delegate::Delegate::push_lock\n             at /workspace/source/tikv/components/cdc/src/delegate.rs:374:21\n      cdc::delegate::Delegate::sink_txn_put\n             at /workspace/source/tikv/components/cdc/src/delegate.rs:1054:41\n      cdc::delegate::Delegate::sink_put\n             at /workspace/source/tikv/components/cdc/src/delegate.rs:1003:13\n      cdc::delegate::Delegate::sink_data\n             at /workspace/source/tikv/components/cdc/src/delegate.rs:871:33\n   9: cdc::delegate::Delegate::on_batch\n             at /workspace/source/tikv/components/cdc/src/delegate.rs:734:17\n  10: cdc::endpoint::Endpoint<T,E,S>::on_multi_batch\n             at /workspace/source/tikv/components/cdc/src/endpoint.rs:999:33\n      <cdc::endpoint::Endpoint<T,E,S> as tikv_util::worker::pool::Runnable>::run\n             at /workspace/source/tikv/components/cdc/src/endpoint.rs:1225:18\n  11: tikv_util::worker::pool::Worker::start_with_timer_impl::{{closure}}\n             at /workspace/source/tikv/components/tikv_util/src/worker/pool.rs:532:25\n      <tracker::tls::TrackedFuture<F> as core::future::future::Future>::poll::{{closure}}\n             at /workspace/source/tikv/components/tracker/src/tls.rs:64:23\n      std::thread::local::LocalKey<T>::try_with\n             at /root/.rustup/toolchains/nightly-2023-12-28-aarch64-unknown-linux-gnu/lib/rustlib/src/rust/library/std/src/thread/local.rs:270:16\n      std::thread::local::LocalKey<T>::with\n             at /root/.rustup/toolchains/nightly-2023-12-28-aarch64-unknown-linux-gnu/lib/rustlib/src/rust/library/std/src/thread/local.rs:246:9\n      <tracker::tls::TrackedFuture<F> as core::future::future::Future>::poll\n             at /workspace/source/tikv/components/tracker/src/tls.rs:62:27\n      <futures_util::future::future::map::Map<Fut,F> as core::future::future::Future>::poll\n             at /workspace/.cargo/registry/src/mirrors.tuna.tsinghua.edu.cn-df7c3c540f42cdbd/futures-util-0.3.31/src/future/future/map.rs:55:37\n      <futures_util::future::future::Map<Fut,F> as core::future::future::Future>::poll\n             at /workspace/.cargo/registry/src/mirrors.tuna.tsinghua.edu.cn-df7c3c540f42cdbd/futures-util-0.3.31/src/lib.rs:86:13\n      yatp::task::future::RawTask<F>::poll\n             at /workspace/.cargo/git/checkouts/yatp-e704b73c3ee279b6/793be4d/src/task/future.rs:59:9\n  12: yatp::task::future::TaskCell::poll\n             at /workspace/.cargo/git/checkouts/yatp-e704b73c3ee279b6/793be4d/src/task/future.rs:103:9\n      <yatp::task::future::Runner as yatp::pool::runner::Runner>::handle\n             at /workspace/.cargo/git/checkouts/yatp-e704b73c3ee279b6/793be4d/src/task/future.rs:387:20\n  13: <tikv_util::yatp_pool::YatpPoolRunner<T> as yatp::pool::runner::Runner>::handle\n             at /workspace/source/tikv/components/tikv_util/src/yatp_pool/mod.rs:199:24\n      yatp::pool::worker::WorkerThread<T,R>::run\n             at /workspace/.cargo/git/checkouts/yatp-e704b73c3ee279b6/793be4d/src/pool/worker.rs:48:13\n      yatp::pool::builder::LazyBuilder<T>::build::{{closure}}\n             at /workspace/.cargo/git/checkouts/yatp-e704b73c3ee279b6/793be4d/src/pool/builder.rs:114:25\n      std::sys_common::backtrace::__rust_begin_short_backtrace\n             at /root/.rustup/toolchains/nightly-2023-12-28-aarch64-unknown-linux-gnu/lib/rustlib/src/rust/library/std/src/sys_common/backtrace.rs:155:18\n  14: std::thread::Builder::spawn_unchecked_::{{closure}}::{{closure}}\n             at /root/.rustup/toolchains/nightly-2023-12-28-aarch64-unknown-linux-gnu/lib/rustlib/src/rust/library/std/src/thread/mod.rs:529:17\n      <core::panic::unwind_safe::AssertUnwindSafe<F> as core::ops::function::FnOnce<()>>::call_once\n             at /root/.rustup/toolchains/nightly-2023-12-28-aarch64-unknown-linux-gnu/lib/rustlib/src/rust/library/core/src/panic/unwind_safe.rs:272:9\n      std::panicking::try::do_call\n             at /root/.rustup/toolchains/nightly-2023-12-28-aarch64-unknown-linux-gnu/lib/rustlib/src/rust/library/std/src/panicking.rs:552:40\n      std::panicking::try\n             at /root/.rustup/toolchains/nightly-2023-12-28-aarch64-unknown-linux-gnu/lib/rustlib/src/rust/library/std/src/panicking.rs:516:19\n      std::panic::catch_unwind\n             at /root/.rustup/toolchains/nightly-2023-12-28-aarch64-unknown-linux-gnu/lib/rustlib/src/rust/library/std/src/panic.rs:142:14\n      std::thread::Builder::spawn_unchecked_::{{closure}}\n             at /root/.rustup/toolchains/nightly-2023-12-28-aarch64-unknown-linux-gnu/lib/rustlib/src/rust/library/std/src/thread/mod.rs:528:30\n      core::ops::function::FnOnce::call_once{{vtable.shim}}\n             at /root/.rustup/toolchains/nightly-2023-12-28-aarch64-unknown-linux-gnu/lib/rustlib/src/rust/library/core/src/ops/function.rs:250:5\n  15: <alloc::boxed::Box<F,A> as core::ops::function::FnOnce<Args>>::call_once\n             at /root/.rustup/toolchains/nightly-2023-12-28-aarch64-unknown-linux-gnu/lib/rustlib/src/rust/library/alloc/src/boxed.rs:2015:9\n      <alloc::boxed::Box<F,A> as core::ops::function::FnOnce<Args>>::call_once\n             at /root/.rustup/toolchains/nightly-2023-12-28-aarch64-unknown-linux-gnu/lib/rustlib/src/rust/library/alloc/src/boxed.rs:2015:9\n      std::sys::unix::thread::Thread::new::thread_start\n             at /root/.rustup/toolchains/nightly-2023-12-28-aarch64-unknown-linux-gnu/lib/rustlib/src/rust/library/std/src/sys/unix/thread.rs:108:17\n  16: start_thread\n  17: thread_start\n","location":"components/cdc/src/delegate.rs:374","thread_name":"cdc-0"}
{"level":"INFO","caller":"lib.rs:91","message":"Welcome to TiKV","time":"2025/05/29 02:44:42.869 +00:00","thread_id":1}
{"level":"INFO","caller":"lib.rs:96","message":"Release Version:   8.5.1","time":"2025/05/29 02:44:42.870 +00:00","thread_id":1}
{"level":"INFO","caller":"lib.rs:96","message":"Edition:           Enterprise","time":"2025/05/29 02:44:42.870 +00:00","thread_id":1}
{"level":"INFO","caller":"lib.rs:96","message":"Git Commit Hash:   25897b73e1f0e93f616c764dbe65ed7db02edd07","time":"2025/05/29 02:44:42.870 +00:00","thread_id":1}
```

```
   0: tikv_util::set_panic_hook::{{closure}}
             at /workspace/source/tikv/components/tikv_util/src/lib.rs:478:18
   1: <alloc::boxed::Box<F,A> as core::ops::function::Fn<Args>>::call
             at /root/.rustup/toolchains/nightly-2023-12-28-aarch64-unknown-linux-gnu/lib/rustlib/src/rust/library/alloc/src/boxed.rs:2029:9
      std::panicking::rust_panic_with_hook
             at /root/.rustup/toolchains/nightly-2023-12-28-aarch64-unknown-linux-gnu/lib/rustlib/src/rust/library/std/src/panicking.rs:783:13
   2: std::panicking::begin_panic_handler::{{closure}}
             at /root/.rustup/toolchains/nightly-2023-12-28-aarch64-unknown-linux-gnu/lib/rustlib/src/rust/library/std/src/panicking.rs:657:13
   3: std::sys_common::backtrace::__rust_end_short_backtrace
             at /root/.rustup/toolchains/nightly-2023-12-28-aarch64-unknown-linux-gnu/lib/rustlib/src/rust/library/std/src/sys_common/backtrace.rs:171:18
   4: rust_begin_unwind
             at /root/.rustup/toolchains/nightly-2023-12-28-aarch64-unknown-linux-gnu/lib/rustlib/src/rust/library/std/src/panicking.rs:645:5
   5: core::panicking::panic_fmt
             at /root/.rustup/toolchains/nightly-2023-12-28-aarch64-unknown-linux-gnu/lib/rustlib/src/rust/library/core/src/panicking.rs:72:14
   6: core::panicking::assert_failed_inner
   7: core::panicking::assert_failed
             at /root/.rustup/toolchains/nightly-2023-12-28-aarch64-unknown-linux-gnu/lib/rustlib/src/rust/library/core/src/panicking.rs:297:5
   8: cdc::delegate::Delegate::push_lock
             at /workspace/source/tikv/components/cdc/src/delegate.rs:374:21
      cdc::delegate::Delegate::sink_txn_put
             at /workspace/source/tikv/components/cdc/src/delegate.rs:1054:41
      cdc::delegate::Delegate::sink_put
             at /workspace/source/tikv/components/cdc/src/delegate.rs:1003:13
      cdc::delegate::Delegate::sink_data
             at /workspace/source/tikv/components/cdc/src/delegate.rs:871:33
   9: cdc::delegate::Delegate::on_batch
             at /workspace/source/tikv/components/cdc/src/delegate.rs:734:17
  10: cdc::endpoint::Endpoint<T,E,S>::on_multi_batch
             at /workspace/source/tikv/components/cdc/src/endpoint.rs:999:33
      <cdc::endpoint::Endpoint<T,E,S> as tikv_util::worker::pool::Runnable>::run
             at /workspace/source/tikv/components/cdc/src/endpoint.rs:1225:18
  11: tikv_util::worker::pool::Worker::start_with_timer_impl::{{closure}}
             at /workspace/source/tikv/components/tikv_util/src/worker/pool.rs:532:25
      <tracker::tls::TrackedFuture<F> as core::future::future::Future>::poll::{{closure}}
             at /workspace/source/tikv/components/tracker/src/tls.rs:64:23
      std::thread::local::LocalKey<T>::try_with
             at /root/.rustup/toolchains/nightly-2023-12-28-aarch64-unknown-linux-gnu/lib/rustlib/src/rust/library/std/src/thread/local.rs:270:16
      std::thread::local::LocalKey<T>::with
             at /root/.rustup/toolchains/nightly-2023-12-28-aarch64-unknown-linux-gnu/lib/rustlib/src/rust/library/std/src/thread/local.rs:246:9
      <tracker::tls::TrackedFuture<F> as core::future::future::Future>::poll
             at /workspace/source/tikv/components/tracker/src/tls.rs:62:27
      <futures_util::future::future::map::Map<Fut,F> as core::future::future::Future>::poll
             at /workspace/.cargo/registry/src/mirrors.tuna.tsinghua.edu.cn-df7c3c540f42cdbd/futures-util-0.3.31/src/future/future/map.rs:55:37
      <futures_util::future::future::Map<Fut,F> as core::future::future::Future>::poll
             at /workspace/.cargo/registry/src/mirrors.tuna.tsinghua.edu.cn-df7c3c540f42cdbd/futures-util-0.3.31/src/lib.rs:86:13
      yatp::task::future::RawTask<F>::poll
             at /workspace/.cargo/git/checkouts/yatp-e704b73c3ee279b6/793be4d/src/task/future.rs:59:9
  12: yatp::task::future::TaskCell::poll
             at /workspace/.cargo/git/checkouts/yatp-e704b73c3ee279b6/793be4d/src/task/future.rs:103:9
      <yatp::task::future::Runner as yatp::pool::runner::Runner>::handle
             at /workspace/.cargo/git/checkouts/yatp-e704b73c3ee279b6/793be4d/src/task/future.rs:387:20
  13: <tikv_util::yatp_pool::YatpPoolRunner<T> as yatp::pool::runner::Runner>::handle
             at /workspace/source/tikv/components/tikv_util/src/yatp_pool/mod.rs:199:24
      yatp::pool::worker::WorkerThread<T,R>::run
             at /workspace/.cargo/git/checkouts/yatp-e704b73c3ee279b6/793be4d/src/pool/worker.rs:48:13
      yatp::pool::builder::LazyBuilder<T>::build::{{closure}}
             at /workspace/.cargo/git/checkouts/yatp-e704b73c3ee279b6/793be4d/src/pool/builder.rs:114:25
      std::sys_common::backtrace::__rust_begin_short_backtrace
             at /root/.rustup/toolchains/nightly-2023-12-28-aarch64-unknown-linux-gnu/lib/rustlib/src/rust/library/std/src/sys_common/backtrace.rs:155:18
  14: std::thread::Builder::spawn_unchecked_::{{closure}}::{{closure}}
             at /root/.rustup/toolchains/nightly-2023-12-28-aarch64-unknown-linux-gnu/lib/rustlib/src/rust/library/std/src/thread/mod.rs:529:17
      <core::panic::unwind_safe::AssertUnwindSafe<F> as core::ops::function::FnOnce<()>>::call_once
             at /root/.rustup/toolchains/nightly-2023-12-28-aarch64-unknown-linux-gnu/lib/rustlib/src/rust/library/core/src/panic/unwind_safe.rs:272:9
      std::panicking::try::do_call
             at /root/.rustup/toolchains/nightly-2023-12-28-aarch64-unknown-linux-gnu/lib/rustlib/src/rust/library/std/src/panicking.rs:552:40
      std::panicking::try
             at /root/.rustup/toolchains/nightly-2023-12-28-aarch64-unknown-linux-gnu/lib/rustlib/src/rust/library/std/src/panicking.rs:516:19
      std::panic::catch_unwind
             at /root/.rustup/toolchains/nightly-2023-12-28-aarch64-unknown-linux-gnu/lib/rustlib/src/rust/library/std/src/panic.rs:142:14
      std::thread::Builder::spawn_unchecked_::{{closure}}
             at /root/.rustup/toolchains/nightly-2023-12-28-aarch64-unknown-linux-gnu/lib/rustlib/src/rust/library/std/src/thread/mod.rs:528:30
      core::ops::function::FnOnce::call_once{{vtable.shim}}
             at /root/.rustup/toolchains/nightly-2023-12-28-aarch64-unknown-linux-gnu/lib/rustlib/src/rust/library/core/src/ops/function.rs:250:5
  15: <alloc::boxed::Box<F,A> as core::ops::function::FnOnce<Args>>::call_once
             at /root/.rustup/toolchains/nightly-2023-12-28-aarch64-unknown-linux-gnu/lib/rustlib/src/rust/library/alloc/src/boxed.rs:2015:9
      <alloc::boxed::Box<F,A> as core::ops::function::FnOnce<Args>>::call_once
             at /root/.rustup/toolchains/nightly-2023-12-28-aarch64-unknown-linux-gnu/lib/rustlib/src/rust/library/alloc/src/boxed.rs:2015:9
      std::sys::unix::thread::Thread::new::thread_start
             at /root/.rustup/toolchains/nightly-2023-12-28-aarch64-unknown-linux-gnu/lib/rustlib/src/rust/library/std/src/sys/unix/thread.rs:108:17
  16: start_thread
  17: thread_start
```

@cfzjywxk looks like those panics that are added for debugging are red-herring. For example, on the apply path, it collects keys in batches through:

`if r.get_cmd_type() == CmdType::Put && r.get_put().get_cf() == CF_LOCK {`

However, there could be PUT-> DELETE -> PUT on the LOCK CF for the same key, so we need to remove from the hashmap if there's a DELETE.

```
[2025/10/03 21:35:01.429 +00:00] [DEBUG] [apply.rs:1492] ["applied command"] [index=3484278] [peer_id=32985418] [region_id=32985417] [thread_id=249]
[2025/10/03 21:35:01.429 +00:00] [DEBUG] [apply.rs:1492] ["applied command"] [index=3484279] [peer_id=32985418] [region_id=32985417] [thread_id=249]
[2025/10/03 21:35:01.429 +00:00] [DEBUG] [apply.rs:661] ["raft log is applied to the kv db"] [cmd_batch=2] [req_info=] [thread_id=249]
[2025/10/03 21:35:01.429 +00:00] [DEBUG] [apply.rs:683] ["[for debug] during apply PUT: key=7480000000000022FF8A5F698000000000FF000002014A6F665AFF596D6572FF000000FF0000000000F70175FF726E3A6C693A62FFFF6F6F747374726170FFFF3A313735393532FF37FF333031333534FF0000FD0000000000FA, CF=\"lock\""] [thread_id=249]
[2025/10/03 21:35:01.429 +00:00] [DEBUG] [apply.rs:683] ["[for debug] during apply PUT: key=7480000000000022FF8A5F698000000000FF000002014A6F665AFF596D6572FF000000FF0000000000F70175FF726E3A6C693A62FFFF6F6F747374726170FFFF3A313735393532FF37FF333031333534FF0000FD0000000000FAF99950003CA3FEC9, CF=\"write\""] [thread_id=249]
[2025/10/03 21:35:01.429 +00:00] [DEBUG] [apply.rs:691] ["[for debug] during apply DELETE: key=7480000000000022FF8A5F698000000000FF000002014A6F665AFF596D6572FF000000FF0000000000F70175FF726E3A6C693A62FFFF6F6F747374726170FFFF3A313735393532FF37FF333031333534FF0000FD0000000000FA, CF=\"lock\""] [thread_id=249]
[2025/10/03 21:35:01.429 +00:00] [DEBUG] [apply.rs:697] ["[for debug] this is a DELETE on Lock CF, should've removed key from map: key=7480000000000022FF8A5F698000000000FF000002014A6F665AFF596D6572FF000000FF0000000000F70175FF726E3A6C693A62FFFF6F6F747374726170FFFF3A313735393532FF37FF333031333534FF0000FD0000000000FA"] [thread_id=249]
[2025/10/03 21:35:01.429 +00:00] [DEBUG] [apply.rs:683] ["[for debug] during apply PUT: key=7480000000000022FF8A5F698000000000FF000002014A6F665AFF596D6572FF000000FF0000000000F70175FF726E3A6C693A62FFFF6F6F747374726170FFFF3A313735393532FF37FF333031333534FF0000FD0000000000FA, CF=\"lock\""] [thread_id=249]
[2025/10/03 21:35:01.429 +00:00] [ERROR] [apply.rs:715] ["[for debug] found duplicate PUT on lock cf: key=7480000000000022FF8A5F698000000000FF000002014A6F665AFF596D6572FF000000FF0000000000F70175FF726E3A6C693A62FFFF6F6F747374726170FFFF3A313735393532FF37FF333031333534FF0000FD0000000000FA, existing_value=50960174800000000000228A5F698000000000000002014A6F665A596D6572FF0000000000000000F70175726E3A6C693A62FF6F6F747374726170FF3A31373539353237FF3330313335340000FDB282F09AFCFFABB306B8177611007D018000020000000102010002000000740000000000000001630666AFFFC35C01336102820174800000000000228A5F6980000000000000030380000199ABFFF0EA0175726E3A6C693A62FF6F6F747374726170FF3A31373539353237FF3330313335340000FD6074800000000000228A5F720175726E3A6C693A62FF6F6F747374726170FF3A31373539353237FF3330313335340000FD, new_value=446074800000000000228A5F720175726E3A6C693A62FF6F6F747374726170FF3A31373539353237FF3330313335340000FDC482F09AFCFFABB306B917660666AFFFC35C0146740000000000000001630666AFFFC35C014A6100 during apply"] [thread_id=249]
```


```
if r.get_cmd_type() == CmdType::Delete && r.get_put().get_cf() == CF_LOCK {
   lock_cf_map.remove(&key.to_vec());
}  
```

--- a/components/cdc/src/delegate.rs
+++ b/components/cdc/src/delegate.rs
@@ -381,9 +381,10 @@ impl Delegate {
                             "region_id" => self.region_id,
                         );
                     }
-                    // There could be stale locks in the lock_tracker due to scenarios such as the overlapped
-                    // write/rollback issue (#18498). We can safely ignore such stale locks, while keeping
-                    // the invariant of monotonically increasing start_ts and generation.
+                    // There could be stale locks in the lock_tracker due to scenarios such as the
+                    // overlapped write/rollback issue (#18498). We can safely
+                    // ignore such stale locks, while keeping the invariant of
+                    // monotonically increasing start_ts and generation.
                     assert!(x.get().ts <= start_ts.ts);
                     assert!(x.get().generation <= start_ts.generation);
                     x.get_mut().ts = start_ts.ts;

--- a/components/cdc/tests/integrations/test_cdc.rs
+++ b/components/cdc/tests/integrations/test_cdc.rs
@@ -12,7 +12,7 @@ use pd_client::PdClient;
 use raft::eraftpb::MessageType;
 use test_raftstore::*;
 use tikv::server::DEFAULT_CLUSTER_ID;
-use tikv_util::{HandyRwLock, debug, error,config::ReadableDuration};
+use tikv_util::{HandyRwLock, config::ReadableDuration, debug, error};
 use txn_types::{Key, Lock, LockType, TimeStamp};
 
 use crate::{TestSuite, TestSuiteBuilder, new_event_feed, new_event_feed_v2};
@@ -2918,13 +2918,14 @@ fn test_cdc_overlapped_write_bug() {
     debug!("region:"; "region_id" => region_id);
 
     let test_key = b"test_key";
-    let primary_key_t2 = b"t2_primary";  // t2's primary (different from test_key)
+    let primary_key_t2 = b"t2_primary"; // t2's primary (different from test_key)
     let value1 = b"value1";
     let value2 = b"value2";
     let value3 = b"value3";
 
     // Phase 1: t1 prewrites and commits test_key
-    // CRITICAL: We need t2's start_ts to equal t1's commit_ts for overlapped write detection!
+    // CRITICAL: We need t2's start_ts to equal t1's commit_ts for overlapped write
+    // detection!
     let t1_start_ts = block_on(suite.cluster.pd_client.get_tso()).unwrap();
     debug!("t1:"; "start_ts" => t1_start_ts);
 
@@ -2944,7 +2945,10 @@ fn test_cdc_overlapped_write_bug() {
     commit_req.set_keys(vec![test_key.to_vec()].into());
     commit_req.set_commit_version(t1_commit_ts.into_inner());
 
-    suite.get_tikv_client(region_id).kv_commit(&commit_req).unwrap();
+    suite
+        .get_tikv_client(region_id)
+        .kv_commit(&commit_req)
+        .unwrap();
     debug!("t1:"; "commit_ts" => t1_commit_ts);
     // CF_WRITE now has t1's committed write
 
@@ -2975,7 +2979,7 @@ fn test_cdc_overlapped_write_bug() {
 
     // Phase 2: t2 prewrites test_key as a SECONDARY key
     // CRITICAL: Use t1's commit_ts as t2's start_ts to trigger overlapped write!
-    let t2_start_ts = t1_commit_ts;  // This is the key: start_ts == commit_ts of t1!
+    let t2_start_ts = t1_commit_ts; // This is the key: start_ts == commit_ts of t1!
     debug!("t2: start_ts is the same as t1's commit_ts!"; "start_ts" => t2_start_ts);
     // t2: primary_key is different from the test_key
 
@@ -3028,7 +3032,7 @@ fn test_cdc_overlapped_write_bug() {
 
     let mut rollback_req = BatchRollbackRequest::default();
     rollback_req.set_context(ctx.clone());
-    rollback_req.set_keys(vec![test_key.to_vec()].into());  // Only rollback test_key
+    rollback_req.set_keys(vec![test_key.to_vec()].into()); // Only rollback test_key
     rollback_req.set_start_version(t2_start_ts.into_inner());
 
     tikv_client.kv_batch_rollback(&rollback_req).unwrap();
@@ -3070,7 +3074,7 @@ fn test_cdc_overlapped_write_bug() {
         let events_result = std::panic::catch_unwind(std::panic::AssertUnwindSafe(|| {
             receive_event(true).events.to_vec()
         }));
-        
+
         if let Ok(events) = events_result {
             for event in events {
                 if let Some(Event_oneof_event::Entries(entries)) = event.event {
@@ -3166,7 +3170,10 @@ fn test_verify_overlapped_write_skips_cf_write() {
     commit_req.set_start_version(t1_start_ts.into_inner());
     commit_req.set_keys(vec![test_key.to_vec()].into());
     commit_req.set_commit_version(t1_commit_ts.into_inner());
-    suite.get_tikv_client(region_id).kv_commit(&commit_req).unwrap();
+    suite
+        .get_tikv_client(region_id)
+        .kv_commit(&commit_req)
+        .unwrap();
 
     // t2: Prewrite with test_key as SECONDARY (primary is different)
     // CRITICAL: Use t1's commit_ts as t2's start_ts!
@@ -3200,7 +3207,8 @@ fn test_verify_overlapped_write_skips_cf_write() {
 
     // t2: Rolled back test_key (secondary)
     // Expected behavior:
-    //   - get_txn_commit_record finds: commit_ts={t1_commit_ts} == start_ts={t2_start_ts}
+    //   - get_txn_commit_record finds: commit_ts={t1_commit_ts} ==
+    //     start_ts={t2_start_ts}
     //   - Overlapped write detected!
     //   - test_key is NOT primary → protected=false
     //   - make_rollback(protected=false, overlapped_write=Some) returns None

